src/app/scopes-dialog/scopes-dialog.component.ts

```diff
@@ -19,8 +19,36 @@ declare let mwfAutoInit;
   templateUrl: './scopes-dialog.component.html',
 })
 export class ScopesDialogComponent extends GraphExplorerComponent implements AfterViewInit {
+  public static fabricDialog: any;
+
+  public static showDialog() { // tslint:disable-line
+
+    const scopesDialog = document.querySelector('#scopes-dialog');
+    ScopesDialogComponent.fabricDialog = new fabric.Dialog(scopesDialog);
+    ScopesDialogComponent.fabricDialog.open();
+
+    mwfAutoInit.ComponentFactory.create([{
+      component: mwfAutoInit.Checkbox,
+    }]);
+
+    // We are explicitly focusing on the close icon button here despite setting the autofocus property.
+    // This is because Edge does not give this element focus even with the autofocus property set.
+    // See: https://stackoverflow.com/questions/51867504/edge-how-to-make-autofocus-work-with-refresh-button
+    (scopesDialog.childNodes[2] as any).focus();
+  }
+
+  public static closeDialog() {
+    ScopesDialogComponent.fabricDialog.close();
```

Comment:
I think we should set the focus back to the button that opened this dialog.

`(document.querySelector('#scopes-dialog') as any).focus();`

Reply:
Moving the focus back to the button that opened the dialog would make it look like the focus is in the background element. As the Dialog will be overlaying the document. Also, the requirement was to close the dialog on pressing escape.

Reply:
Maybe my example is wrong. 

Once the the ScopesDialog is closed, we should have the tab focus back on the element (button I think) that was used to open the ScopesDialog. It wouldn't be in the background since the ScopesDialog would already be closed. That way tabbing order is maintained. What is the behavior when the ScopesDialog is closed? Does the tab order continue from ScopesDialog?

Reply:
Yes, the tab order continues from modify permissions but this is accidental as the `modify permissions` button is also the first element with tab focus in the document. I am adding your suggestion to make it explicit.